code/src/cljs/sixsq/nuvla/ui/utils/general.cljs

```diff
@@ -380,3 +380,30 @@
 (defn envsubst-str
   [envsubst s]
   (str/replace s (->> envsubst keys (map regex-escape) (str/join "|") re-pattern) envsubst))
+
+(defn read-fav-language!
+  "Reads js/window.navigator.languages and returns first element or 'en-US'."
+  []
+  (-> js/window
+      .-navigator
+      .-languages
+      first
+      (or "en-US")))
+
+(defn format-number
+  ([amount]
+   (format-number amount {:locale (read-fav-language!)}))
+  ([amount {:keys [locale style currency]
+            :or {locale   (read-fav-language!)
```

Comment:
read-fav-language! is called at each format. Why not create a def that store the browser language?

Reply:
Because it looks like I'm still a react/Javascript dev  :)

Makes sense, will change